pytorch_lightning/trainer/trainer.py

```diff
@@ -2010,6 +2010,23 @@ def should_rank_save_checkpoint(self) -> bool:
     def num_nodes(self) -> int:
         return getattr(self.strategy, "num_nodes", 1)
 
+    @property
+    def device_ids(self) -> List[int]:
```

Comment:
I would have preferred if this was implemented on the accelerator connector. It is IMO not the responsibility of the Trainer to compute the device_ids off the strategy. Note the the strategy is owned by the connector, not the trainer. As the trainer has already many responsibilities, we should delegate as much as possible to the concrete components.

Reply:
Thanks for reviewing @awaelchli!

My personal thoughts: the accelerator connector should stay internal, and accelerator connector should not expose those device related public properties, since they are easily computed off the strategy. 

> Note the the strategy is owned by the connector, not the trainer. 

Personally I feel the accelerator connector does a lot of checks/set up/initialization and configuration of the accelerator and strategy. After things are set up already, strategy/accelerator are already accessible on the Trainer. 

I think delegating computations using accelerator/strategy to the accelerator_connector will make us add many public properties to the internal-facing accelerator_connector. Deriving directly from strategy/accelerator directly are more straight forward, and they(strategy, accelerator) are very external facing as well. 

I'm very new to this and my feelings might be wrong. Feel free to add more thoughts to issue #12171!